contributors/design-proposals/storage/container-storage-interface.md

```diff
@@ -410,35 +410,42 @@ When the controller decides to detach a CSI volume, it will call the in-tree CSI
 
 ### Recommended Mechanism for Deploying CSI Drivers on Kubernetes
 
-Although, Kubernetes does not dictate the packaging for a CSI volume driver, it offers the following recommendations to simplify deployment of a containerized CSI volume driver on Kubernetes.
+Although Kubernetes does not dictate the packaging for a CSI volume driver, it offers the following recommendations to simplify deployment of a containerized CSI volume driver on Kubernetes.
 
 ![Recommended CSI Deployment Diagram](container-storage-interface_diagram1.png?raw=true "Recommended CSI Deployment Diagram")
 
 To deploy a containerized third-party CSI volume driver, it is recommended that storage vendors:
 
   * Create a “CSI volume driver” container that implements the volume plugin behavior and exposes a gRPC interface via a unix domain socket, as defined in the CSI spec (including Controller, Node, and Identity services).
-  * Bundle the “CSI volume driver” container with helper containers (external-attacher, external-provisioner, Kubernetes CSI Helper) that the Kubernetes team will provide (these helper containers will assist the “CSI volume driver” container in interacting with the Kubernetes system). More specifically, create the following Kubernetes objects:
+  * Bundle the “CSI volume driver” container with helper containers (external-attacher, external-provisioner, node-driver-registrar, cluster-driver-registrar, external-resizer, external-snapshotter, livenessprobe) that the Kubernetes team will provide (these helper containers will assist the “CSI volume driver” container in interacting with the Kubernetes system). More specifically, create the following Kubernetes objects:
     * A `StatefulSet` (to facilitate communication with the Kubernetes controllers) that has:
```

Comment:
@jsafrane I believe you instead recommend a Deployment with leader election?

Because if a node becomes unresponsive, then the StatefulSet pod will never be deleted and recreated.

Reply:
Yes, I recommend Deployment with >1 replicas with leader election. As Michelle wrote, StatefulSet can be painfully slow when a node becomes unavailable / broken.